File: CPP/kelas.cpp
/*Saya Muhammad Rafie Alhabsyi Setiawan NIM 2202400
mengerjakan soal Latihan Praktikum 1
dalam mata kuliah DESAIN DAN PEMROGRAMAN BERORIENTASI
untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

#include <iostream>
#include <vector>
#include <string>

using namespace std;

class AnggotaDPR {
private:
    int id;
    string nama;
    string bidang;
    string partai;

public:
    AnggotaDPR(int _id, const string& _nama, const string& _bidang, const string& _partai) :
        id(_id), nama(_nama), bidang(_bidang), partai(_partai) {}

    int getId() const { return id; }
    string getNama() const { return nama; }
    string getBidang() const { return bidang; }
    string getPartai() const { return partai; }

    void setNama(const string& _nama) { nama = _nama; }
    void setBidang(const string& _bidang) { bidang = _bidang; }
    void setPartai(const string& _partai) { partai = _partai; }
};

class DaftarAnggotaDPR {
private:
    vector<AnggotaDPR> anggotaDPRs;

public:
    void tambahAnggota(int id, const string& nama, const string& bidang, const string& partai) {
        anggotaDPRs.emplace_back(id, nama, bidang, partai);
    }

    void tampilkanAnggota() const {
        cout << "Daftar Anggota DPR:\n";
        for (const auto& anggota : anggotaDPRs) {
            cout << "ID: " << anggota.getId() << ", Nama: " << anggota.getNama() << ", Bidang: " << anggota.getBidang() << ", Partai: " << anggota.getPartai() << endl;
        }
    }

    void ubahAnggota(int id, const string& nama, const string& bidang, const string& partai) {
        for (auto& anggota : anggotaDPRs) {
            if (anggota.getId() == id) {
                anggota.setNama(nama);
                anggota.setBidang(bidang);
                anggota.setPartai(partai);
                return;
            }
        }
        cout << "Anggota dengan ID " << id << " tidak ditemukan.\n";
    }

    void hapusAnggota(int id) {
        for (auto it = anggotaDPRs.begin(); it != anggotaDPRs.end(); ++it) {
            if (it->getId() == id) {
                anggotaDPRs.erase(it);
                cout << "Anggota dengan ID " << id << " telah dihapus.\n";
                return;
            }
        }
        cout << "Anggota dengan ID " << id << " tidak ditemukan.\n";
    }
};

File: PYTHON/kelas.py
class AnggotaDPR:
    def __init__(self, _id, _nama, _bidang, _partai):
        self.id = _id
        self.nama = _nama
        self.bidang = _bidang
        self.partai = _partai

    def getId(self):
        return self.id

    def getNama(self):
        return self.nama

    def getBidang(self):
        return self.bidang

    def getPartai(self):
        return self.partai

    def setNama(self, _nama):
        self.nama = _nama

    def setBidang(self, _bidang):
        self.bidang = _bidang

    def setPartai(self, _partai):
        self.partai = _partai

class DaftarAnggotaDPR:
    def __init__(self):
        self.anggotaDPRs = []

    def tambahAnggota(self, id, nama, bidang, partai):
        self.anggotaDPRs.append(AnggotaDPR(id, nama, bidang, partai))

    def tampilkanAnggota(self):
        print("Daftar Anggota DPR:")
        for anggota in self.anggotaDPRs:
            print("ID:", anggota.getId(), ", Nama:", anggota.getNama(), ", Bidang:", anggota.getBidang(), ", Partai:", anggota.getPartai())

    def ubahAnggota(self, id, nama, bidang, partai):
        for anggota in self.anggotaDPRs:
            if anggota.getId() == id:
                anggota.setNama(nama)
                anggota.setBidang(bidang)
                anggota.setPartai(partai)
                return
        print("Anggota dengan ID", id, "tidak ditemukan.")

    def hapusAnggota(self, id):
        for anggota in self.anggotaDPRs:
            if anggota.getId() == id:
                self.anggotaDPRs.remove(anggota)
                print("Anggota dengan ID", id, "telah dihapus.")
                return
        print("Anggota dengan ID", id, "tidak ditemukan.")
